webapp/src/main/java/com/box/l10n/mojito/rest/repository/BranchStatisticWS.java

```diff
@@ -52,41 +44,70 @@ public class BranchStatisticWS {
+    Page<Long> branchStatisticIds = sparseBranchStatisticRepository.findAllWithIdOnly(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter)))
+            , pageable);
 
-    // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1
-    // queries
-    List<BranchStatistic> branchStatistics =
-        branchStatisticRepository.findByIdIn(branchStatisticIds.getContent(), pageable.getSort());
+    // Retrieve the Branch Statistic Ids which have TmTextUnit count less than the 'totalCountLte'
+    Page<Long> branchStatisticIdsWithLessThan = sparseBranchStatisticRepository.findAllWithIdOnly(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter))).and(
+                    totalCountLessThanOrEqualsTo(totalCountLte))
+            , pageable);
 
+    // Retrieve the branch statistics where the total text unit count is greater than the limit,
+    // by filtering the branchStatisticIds less than the limit.
+    Page<BranchStatistic> branchStatisticsWithoutTextUnits = branchStatisticRepository.findAll(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    totalCountGreaterThan(totalCountLte))
+            , pageable);
+
+    // Set the BranchTextUnitStatistics to null as the branch has a large number of text Units.
+    branchStatisticsWithoutTextUnits.forEach(
+            branchStatistic -> {branchStatistic.setBranchTextUnitStatistics( null );});
+
+
+    // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1 queries
+    List<BranchStatistic> branchStatistics = branchStatisticRepository.findByIdIn(
```

Comment:
Won't this cause the branchStatisticsWithoutTextUnits entities to get their properties using N+1 queries? 
They still have data that needs to be serialized and sent down in JSON form (outside of the TextUnit information that we don't want for them) - except now they won't be retrieved in one go.

Reply:
This line is using the branchStatisticIdsWithLessThan, so not sure why the branchStatisticsWithoutTextUnits would be retrieved here?

Reply:
The problem is that the branchStatisticsWithoutTextUnits entities still have linked entities - if they're not retrieved explicitly through an optimized entity graph - how many queries will they generate to get their related Repository, User, Authority etc. entities?  

Tracking the SQL queries generated should confirm how many queries would result as a result of the response being serialized.